src/master-playlist-controller.js

```diff
@@ -58,31 +60,61 @@ const shouldSwitchToMedia = function({
     return false;
   }
 
+  const sharedLogLine = `allowing switch ${currentPlaylist && currentPlaylist.id || 'null'} -> ${nextPlaylist.id}`;
```

Comment:
Added tons of logging throughout this function so that we can track what is going on by turning debug logging on.